src/components/userForm/userForm.component.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+	createUserAccount,
+	signInUser,
+} from "../../utils/firebase/firebase.utils"
+import {useForm} from "react-hook-form"
+import {useNavigate} from "react-router-dom"
+
+import "./userForm.css"
+
+type UserFormProps = {
+	isLogin: boolean
+}
+
+const UserForm = (props: UserFormProps) => {
+	const {isLogin} = props
+	const navigate = useNavigate()
+
+	const {
+		register,
+		handleSubmit,
+		getValues,
+		//formState: {errors},
+	} = useForm({
+		defaultValues: {
+			name_input: "",
+			email_input: "",
+			password_input: "",
+			confirm_password_input: "",
+		},
+	})
+
+	const onSubmitHandler = handleSubmit(async (data) => {
+		const name = data.name_input
+		const email = data.email_input
+		const password = data.password_input
+		const confirmPassword = data.confirm_password_input
+
+		console.log(name, email, password, confirmPassword)
+		let resp
+		if (isLogin) resp = await signInUser(email, password)
+		else resp = await createUserAccount(email, password)
+
+		if (!(resp instanceof Error)) navigate("/home")
+	})
+
+	return (
+		<div className="formContainer">
+			<form className="formElement" onSubmit={onSubmitHandler}>
+				{!isLogin && (
+					<input
+						{...register("name_input", {required: "Name is required."})}
```

Comment:
It's better to use camelCase for variables.

```suggestion
						{...register("nameInput", {required: "Name is required."})}
```
This goes for all the variables.

Reply:
Fixed.